src/main/webapp/store/results.js

```diff
@@ -0,0 +1,32 @@
+import { combineReducers } from 'redux'
+import { Set } from 'immutable'
+
+const ns = 'ddf.results.reducer'
+const getLocalState = state => state[ns]
+
+export const setSelected = selected => {
+  if (typeof selected === 'string') {
+    return {
+      type: `${ns}/SET_SELECTED`,
+      selected: Set([selected]),
+    }
+  } else {
+    return {
+      type: `${ns}/SET_SELECTED`,
+      selected: Set(selected),
+    }
+  }
+}
+
+const selection = (state = Set(), action) => {
+  switch (action.type) {
+    case `${ns}/SET_SELECTED`:
+      return action.selected
+    default:
+      return state
+  }
+}
+
+export const getSelected = state => getLocalState(state).selection
```

Comment:
would it be possible to simplify this doing something like what we do in https://github.com/andrewkfiedler/reveal/blob/master/src/main/webapp/store/properties.js#L65
and 
https://github.com/andrewkfiedler/reveal/blob/master/src/main/webapp/about.js#L86

Reply:
This is some better generalization of that - ask @djblue for details.